ambry-mysql/src/main/java/com/github/ambry/accountstats/AccountStatsMySqlStore.java

```diff
@@ -334,4 +360,52 @@ private void tryUpdateStorageUsage(short partitionId, short accountId, short con
     } catch (Exception e) {
     }
   }
+
+  /**
+   * A wrapper to the {@link AccountReportsDao.StorageBatchUpdater} and an implementation of {@link ContainerUsageFunction}.
+   */
+  class ContainerStorageFunctionBatchUpdater implements ContainerUsageFunction {
+    private AccountReportsDao.StorageBatchUpdater batch = null;
+    private boolean exceptionCaught = false;
+
+    /**
+     * Constructor to instantiate {@link ContainerStorageFunctionBatchUpdater}.
+     */
+    public ContainerStorageFunctionBatchUpdater() {
+      try {
+        batch = accountReportsDao.new StorageBatchUpdater(config.updateBatchSize);
+      } catch (SQLException e) {
+        // the exception is already logged
+        exceptionCaught = true;
```

Comment:
I don't like this approach of eating/ignoring exceptions everywhere.  I realize you are calling it from a method that doesn't have checked exceptions (and might be difficult to add it), but this code does not look clean.

Reply:
restructure code a bit, it should be good now.